kore/src/Kore/Simplify/Predicate.hs

```diff
@@ -327,13 +330,46 @@ normalizeTop _ = pure (MultiOr.singleton MultiAnd.top)
 simplifyNot ::
     forall simplifier sort.
     Monad simplifier =>
+    MonadSimplify simplifier =>
+    SideCondition RewritingVariableName ->
     Not sort NormalForm ->
     simplifier NormalForm
-simplifyNot Not{notChild = multiOr, notSort} = do
-    disjunctiveNormalForms <- Logic.observeAllT $ do
-        multiAnd <- Logic.scatter multiOr
-        normalizeNotAnd Not{notSort, notChild = multiAnd} & lift
-    normalizeMultiAnd (MultiAnd.make disjunctiveNormalForms)
+simplifyNot sideCondition Not{notChild = multiOr, notSort} = do
+    -- try user-defined rules first
+    mbResult <- runMaybeT applyUserDefined
+    maybe standardSimplification pure mbResult
+  where
+    applyUserDefined :: MaybeT simplifier NormalForm
+    applyUserDefined = do
+        -- produce a single termlike that we can use for matching
+        let arg :: TermLike RewritingVariableName
+            arg =
+                Pattern.toTermLike
+                    . OrPattern.toPattern helpSort
+                    . toOrPattern helpSort
+                    $ multiOr
+            -- HAAAACK: sort stripped in NormalForm of predicates
```

Comment:
🤣

Reply:
why cant `notSort` be used here btw?

Reply:
I was trying to use `notSort` before, but its type is _variable_ in `simplifyNot`. 
Predicates do not have sort information: https://github.com/runtimeverification/haskell-backend/blob/master/kore/src/Kore/Internal/Predicate.hs#L156..L169
Not 100% sure but `notSort` might always be `()` in the call sites of `simplifyNot`.

Reply:
I have refactored the types in this module to avoid confusion. Indeed `simplifyNot` (and many other functions) are only ever called with `sort ~ ()`.

Reply:
is this because a predicate always has the sort bool?